src/hotspot/cpu/x86/x86.ad

```diff
@@ -5775,6 +5775,46 @@ instruct evminmaxFP_reg_eavx(vec dst, vec a, vec b, vec atmp, vec btmp, kReg ktm
   ins_pipe( pipe_slow );
 %}
 
+// --------------------------------- Signum ---------------------------
+
+instruct signumF_reg(regF dst, regF zero, regF one, rFlagsReg cr) %{
+  match(Set dst (SignumF dst (Binary zero one)));
+  effect(KILL cr);
+  format %{ "signumF $dst, $dst" %}
+  ins_encode %{
+    Label exit;
+
+    __ ucomiss($dst$$XMMRegister, $zero$$XMMRegister);
+    __ jcc(Assembler::parity, exit);
+    __ jcc(Assembler::equal, exit);
+    __ movflt($dst$$XMMRegister, $one$$XMMRegister);
+    __ jcc(Assembler::above, exit);
+    __ movflt($dst$$XMMRegister, $zero$$XMMRegister);
```

Comment:
Is it possible to use just one instruction to assign -1 to $dst?

Maybe, you can try to follow negF_reg/negF_reg_reg.

Reply:
Good idea. Changed to use `xorps($dst$$XMMRegister, ExternalAddress(float_signflip()));` and `xorpd($dst$$XMMRegister, ExternalAddress(double_signflip()));`